presto-pinot/src/main/java/io/prestosql/pinot/query/PinotQueryBuilder.java

```diff
@@ -71,15 +72,13 @@ public static String generatePql(PinotTableHandle tableHandle, List<PinotColumnH
                 .append(getTableName(tableHandle, tableNameSuffix))
                 .append(" ");
         generateFilterPql(pqlBuilder, tableHandle, timePredicate, columnHandles);
-        OptionalLong limit = tableHandle.getLimit();
-        if (limit.isPresent()) {
-            pqlBuilder.append(" LIMIT ")
-                    .append(limit.getAsLong());
-        }
-        else {
-            pqlBuilder.append(" LIMIT ")
-                    .append(Integer.MAX_VALUE);
+        OptionalLong appliedLimit = tableHandle.getLimit();
+        long limit = limitForSegmentQueries;
```

Comment:
If I understand correctly, when the config property is set, all queries will get a LIMIT injected. Is that right? 

If so, this is problematic, as it might cause queries to silently return fewer results than the user expected. The correct behavior would be to fail if the query produces more rows than the configured limit. To do so, we can inject `limit + 1`, and if the count of rows returns equals `limit + 1`, fail the query.

Reply:
Sounds good, have to do this per split as there is no way that I'm aware of to keep track of the global rowcount from the page source.